packages/crawler/src/crawler/simple-crawler-engine.spec.ts

```diff
@@ -2,28 +2,164 @@
+
     let testSubject: SimpleCrawlerEngine;
     let crawlerRunOptions: CrawlerRunOptions;
 
-    beforeAll(() => {
-        testSubject = new SimpleCrawlerEngine();
+    const localOutputDir = 'local output dir';
+    const maxRequestsPerCrawl = 10;
+    const pageStub = {} as Page;
+    const discoveryPatterns = ['discovery pattern'];
+    const crawlResults = ['url1', 'url2'];
+    const requestProcessorStub = {
+        handleRequest: () => null,
+        handleFailedRequest: () => null,
+        getResults: () => crawlResults,
+    } as RequestProcessor;
+
+    beforeEach(() => {
+        requestQueueStub = {} as Apify.RequestQueue;
+        requestQueueProviderMock = Mock.ofInstance(() => null);
+        crawlerFactoryMock = Mock.ofType<CrawlerFactory>();
+        crawlerConfigurationMock = Mock.ofType<CrawlerConfiguration>();
+        basicCrawlerMock = Mock.ofType<Apify.BasicCrawler>();
+        enqueueLinksExtMock = Mock.ofType<typeof Apify.utils.enqueueLinks>();
+
+        testSubject = new SimpleCrawlerEngine(
+            requestQueueProviderMock.object,
+            crawlerFactoryMock.object,
+            crawlerConfigurationMock.object,
+            requestProcessorStub,
+            enqueueLinksExtMock.object,
+        );
         crawlerRunOptions = {
-            localOutputDir: 'localOutputDir',
+            localOutputDir: localOutputDir,
             memoryMBytes: 100,
             silentMode: true,
             debug: false,
-        } as CrawlerRunOptions;
+            page: pageStub,
+        } as SimpleCrawlerRunOptions;
+    });
+
+    afterEach(() => {
+        crawlerConfigurationMock.verifyAll();
+        crawlerFactoryMock.verifyAll();
+        requestQueueProviderMock.verifyAll();
+        basicCrawlerMock.verifyAll();
+        enqueueLinksExtMock.verifyAll();
     });
 
-    it('returns list of urls', async () => {
-        const expectedUrls: string[] = [];
+    it.each([true, false])('returns list of urls with debug = %s', async (debug) => {
+        crawlerRunOptions.debug = debug;
+        setupCrawlerConfig();
+        if (debug) {
+            crawlerConfigurationMock.setup((cc) => cc.setSilentMode(false)).verifiable();
+        }
+        requestQueueProviderMock
+            .setup((rqp) => rqp())
+            .returns(() => Promise.resolve(requestQueueStub))
+            .verifiable();
+
+        const expectedCrawlOptions = getCrawlerOptions(debug);
+        crawlerFactoryMock
+            .setup((cf) => cf.createBasicCrawler(It.isObjectWith(expectedCrawlOptions)))
+            .returns(() => basicCrawlerMock.object)
+            .verifiable();
+        basicCrawlerMock.setup((bc) => bc.run()).verifiable();
+        setupEnqueueLinks();
+
+        const expectedUrls: string[] = crawlResults;
 
         const discoveredUrls = await testSubject.start(crawlerRunOptions);
 
         expect(discoveredUrls).toEqual(expectedUrls);
     });
+
+    it.skip('run e2e website crawl', async () => {
```

Comment:
was .skip purposly left in the PR?

Reply:
Yes, this test creates all the dependencies of the crawler instead of mocking them and runs it end-to-end on a real site. It's marked as skip because we probably don't want to do this in the build/test pipeline, but I left the code in for manual local testing if we need it.